src/pybind/mgr/devicehealth/module.py

```diff
@@ -536,20 +536,37 @@ def _get_device_metrics(self, devid: str,
                             min_sample: Optional[str] = None) -> Dict[str, Dict[str, Any]]:
         res = {}
 
-        SQL = """
+        SQL_EXACT = """
+        SELECT time, raw_smart
+            FROM DeviceHealthMetrics
+            WHERE devid = ? AND time = ?
+            ORDER BY time DESC;
+        """
+        SQL_MIN = """
         SELECT time, raw_smart
             FROM DeviceHealthMetrics
-            WHERE devid = ? AND (time = ? OR ? <= time)
+            WHERE devid = ? AND ? <= time
             ORDER BY time DESC;
         """
 
-        isample = self._t2epoch(sample)
-        imin_sample = self._t2epoch(min_sample)
+        isample = None
+        imin_sample = None
+        try:
+            if sample:
```

Comment:
The issue here seems to be that sample is `''`. I would say that's not a valid input and the diskprediction module should be fixed:

https://github.com/ceph/ceph/blob/87ab4d7afa701927dad6084c820a9a22e38397e6/src/pybind/mgr/diskprediction_local/module.py#L129-L130

Or we could change `_t2epoch` to accept the empty string and return 0.